src/main/java/com/team766/controllers/CanSparkMaxSmartMotionRotationalPIDController.java

```diff
@@ -0,0 +1,218 @@
+
+				mc.setSensorPosition(relEncoder);
+				
+
+			}catch (IllegalArgumentException ill){
+				throw new RuntimeException("Error instantiating the PID controller: " + ill);
+			}
+
+		
+	}
+
+	private double absToEu(double abs){
+		return encoderUnitsPerOneAbsolute / (1/abs); 
+	}
+
+	private double euToDegrees(double eu){
+		return eu * degreesToEncoderUnitsRatio;
+	}
+
+	private double degreesToEu(double degrees){
+		return (1/degreesToEncoderUnitsRatio) * degrees;
+	}
+
+	//manually changing the state
+	public void updateState(PIDSTATE state){
+		theState = state;
+	}
+	//changing all PID values at once
+	public void setPIDF(double p, double i, double d, double ff){
+		pid.setP(p);
+		pid.setI(i);
+		pid.setD(d);
+		pid.setFF(ff);
+	}
+	//changing the P value
+	public void setP(double p){
+		pid.setP(p);
+	}
+	//changing the I value
+	public void setI(double i){
+		pid.setI(i);
+	}
+	//changing the D value
+	public void setD(double d){
+		pid.setD(d);
+	}
+	//changing the FF value
+	public void setFf(double ff){
+		pid.setFF(ff);
+	}
+	
+	/*
+	* Here we set the antigrav constant
+ 	* If the mechanism is rotational, this is the amount we multiply the Sine of the sensor position with
+  	* If the mechanism isn't rotational, this is just the amount of power to apply.
+   	* @param k the value to set according to the above condition
+	*/
+	public void setAntigravConstant(double k){
+		antiGravK = k;
+	}
+
+	private void antigrav(){
+		mc.set(antiGravK * Math.sin(euToDegrees(mc.getSensorPosition())));
+	}
+
+	//changing the deadzone
+	public void setDeadzone(double dz){
+		deadzone = dz;
+	}
+	//changing the output range of the speed of the motors
+	public void setOutputRange(double min, double max){
+		pid.setOutputRange(min, max);
+	}
+	//changing the neutral mode of the motor (brake/coast)
+	public void setMotorMode(NeutralMode mode){
+		mc.setNeutralMode(mode);
+	}
+	//setting the maximum and minimum locations that the motor can go to
+	public void setMinMaxLocation(double min, double max){
+		maxPos = max;
+		minPos = min;
+	}
+	//setting the maximum velocity of the motor
+	public void setMaxVel(double max){
+		pid.setSmartMotionMaxVelocity(max, 0);
+		pid.setSmartMotionMinOutputVelocity(0, 0);
+	}
+	//setting the maximum acceleration of the motor
+	public void setMaxAccel(double max){
+		pid.setSmartMotionMaxAccel(max, 0);
+	}
+
+	//1st step to go to a position using the normal PID, setting what you want the position to be
+	public void setSetpoint(double positionInDegrees){
+		setPointPosition = MathUtil.clamp(degreesToEu(positionInDegrees), minPos, maxPos);
+		theState = PIDSTATE.PID;
+	}
+
+	//Failsafe
+	public void stop(){
+		setPointPosition = mc.getSensorPosition();
+		theState = PIDSTATE.OFF;
+	}
+
+	//run loop that actually runs the PID 
+	//You need to call this function repedatly in OI as often as possible
+	public void run(boolean enabled){
+		if(enabled){
```

Comment:
what's the use case for the `enabled` parameter? it seems like the caller could just have `if (enabled) { csmsmrpid.run(); }`